bluemira/base/components.py

```diff
@@ -224,6 +315,7 @@ def add_children(self, children: List[Component], merge_trees=False):
         self: Component
             This component.
         """
+        # rv
```

Comment:
Sorry I put this here because I wanted to ask about the next lines:

```python
        if not isinstance(children, list) or len(children) == 0:
            child = children[0] if isinstance(children, list) else children
            return self.add_child(child)
```

I know this is minor but I think they can be improved.

Because:
1. If children is not a list, the condition is met and child will be assigned to children, but it's not checked if it's a Component
2. If it is a list, and it is empty, the condition is met and children[0] will be accessed but it's empty so will fail.

i.e. add_children([]) and add_children("fail") throw but they aren't handled well.

Maybe I'll just write my suggested change in the pull request.

Reply:
No problem, I'm happy with the change just need to fix the typing of the function too (which was wrong to begin with). I think its now something like:
```
def add_children(self, children: Optional[Union[Component, List[Component]]], merge_trees: bool =False):
```